### PR TITLE
Improve: install man-page and track content of ocamlformat --help

### PR DESCRIPTION
**Pull request by hhugo**

### CODE REVIEW

**Comment by samoht** [comment]
opam is generating and installing its man page from --help here: https://github.com/ocaml/opam/blob/master/doc/Makefile#L30-L41

File: Makefile [hunk 1]
```diff
@@ -22,7 +22,7 @@ setup: dune-workspace
 
 .PHONY: exe
 exe: setup
-	dune build _build/dev/src/ocamlformat.exe _build/release/src/ocamlformat.exe
+	dune build _build/dev/src/ocamlformat.exe _build/release/src/ocamlformat.exe _build/release/src/ocamlformat-help.txt
```
Comment (jberdine) [inline Makefile]: I'm not sure this should be added to the `exe` target. What about adding a `man` or `doc` or something target, and adding that target to `default`?

File: src/dune [hunk 2]
```diff
@@ -20,6 +20,17 @@ let preprocess =
  %s
  (libraries cmdliner ocamlformat_support import ocaml-migrate-parsetree unix))
 
+(rule
+  (targets ocamlformat.1)
+  (deps ocamlformat.exe)
+  (action (with-stdout-to %%{targets} (run ./ocamlformat.exe --help=groff))))
+
+(rule
+  (targets ocamlformat-help.txt)
```
Comment (jberdine) [inline src/dune]: Will dune flip its shit if this is `../ocamlformat-help.txt` so that the docs aren't in the source dir?

Comment (hhugo) [reply src/dune]: `../ocamlformat-help.txt` will probably not work. but we can move that rule to the parent directory.

**Comment by hhugo** [comment]
@jberdine, what do you think about merging this one first before the other PRs touching the Conf.ml ?

**Comment by hhugo** [comment]
I do not know. @diml ? is it possible ?

**Pull request merged**